frontend/ui/widget/imagewidget.lua

```diff
@@ -333,7 +334,19 @@ function ImageWidget:_render()
     elseif self.scale_factor ~= 1 then
         -- scale by scale_factor (not needed if scale_factor == 1)
         logger.dbg("ImageWidget: scaling by", self.scale_factor)
-        self._bb = RenderImage:scaleBlitBuffer(self._bb, bb_w * self.scale_factor, bb_h * self.scale_factor, self._bb_disposable)
+        local _bb = RenderImage:scaleBlitBuffer(self._bb, bb_w * self.scale_factor, bb_h * self.scale_factor, self._bb_disposable)
+        if _bb then
+            self._bb = _bb
+        else
+            self._bb = RenderImage:renderImageFile("resources/icons/src/appbar.magnify.zoom.svg", false, 200, 200)
+
+            local InfoMessage = require("ui/widget/infomessage")
+            UIManager:show(InfoMessage:new{
+                show_icon = true,
+                text = _("Maximum zoom reached. Please zoom out again."),
+            })
+        end
+
```

Comment:
The blank lines are a bit uneeded (they are used in the outer context to separate steps - no need to add them here as there are not really steps of processing).

Also, I'm not sure it's really needed for such a low level ImageWidget to use UIManager:show(InfoMessage).
I would either:
- have a .svg with that message or symbol :)
- or just keep using any previous self._bb : the user will continue trying to zoom in and see nothing changeing, and will guess maximum zoom is reached.

If going with the first solution, cleaner to not use local _bb:
```diff
         self._bb = RenderImage...
+        if not self._bb then
+           -- couldn't make a bb: probably not enough memory for such a large bb
+           self._bb = RenderImage:renderImageFile("resources/icons/src/appbar.magnify.zoom.svg", false, 200, 200)
+        end
```

Reply:
> Also, I'm not sure it's really needed for such a low level ImageWidget to use UIManager:show(InfoMessage). I would either:
> 
>     * have a .svg with that message or symbol :)

Good point.

>     * or just keep using any previous self._bb : the user will continue trying to zoom in and see nothing changeing, and will guess maximum zoom is reached.
 
Would be possible, but as there is no feedback, that maximum zoom is reached, the user could try to zoom in further. The zoom factor gets bigger and bigger with each try. If the user decides to zoom out again, on the first zoom out nothing happens ...

So I think it is better to have a message (svg-image with the message in).


> 
> If going with the first solution, cleaner to not use local _bb:
> 
> ```diff
>          self._bb = RenderImage...
> +        if not self._bb then
> +           -- couldn't make a bb: probably not enough memory for such a large bb
> +           self._bb = RenderImage:renderImageFile("resources/icons/src/appbar.magnify.zoom.svg", false, 200, 200)
> +        end
> ```

OK

Reply:
Yeah, the "no visible feedback" thing was what made me tweak the onZoom handlers because it was confusing as hell in practice.